File: simple-demo/csharp/SimpleDemo.cs
// Kledo API Demo - C# (.NET 6+)
// Jalankan: dotnet run
// Atau: csc SimpleDemo.cs && SimpleDemo.exe

using System.Net.Http.Headers;
using System.Text.Json;

var apiHost = "http://xxx.api.kledo.com/api/v1";
var accessToken = "your_token_here";

using var client = new HttpClient();
client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);

var res = await client.GetAsync($"{apiHost}/finance/accounts?sort_by=name&order_by=asc&per_page=5");
var json = await res.Content.ReadAsStringAsync();
var doc = JsonDocument.Parse(json);
Console.WriteLine(JsonSerializer.Serialize(doc.RootElement, new JsonSerializerOptions { WriteIndented = true }));

File: simple-demo/dart/simple-demo.dart
// Kledo API Demo - Dart
// Jalankan: dart run simple-demo.dart

import 'dart:convert';
import 'dart:io';

void main() async {
  const apiHost = 'http://xxx.api.kledo.com/api/v1';
  const accessToken = 'your_token_here';

  final client = HttpClient();
  final req = await client.getUrl(Uri.parse(
      '$apiHost/finance/accounts?sort_by=name&order_by=asc&per_page=5'));
  req.headers.set('Accept', 'application/json');
  req.headers.set('Authorization', 'Bearer $accessToken');

  final res = await req.close();
  final body = await res.transform(utf8.decoder).join();
  final json = jsonDecode(body);
  print(const JsonEncoder.withIndent('  ').convert(json));
}
